packages/perseus/src/components/image-loader.tsx

```diff
@@ -41,7 +41,7 @@ type Props = {
     // When the DOM updates to replace the preloader with the image, or
     // vice-versa, we trigger this callback.
     onUpdate: (status: (typeof Status)[keyof typeof Status]) => void;
-    preloader: () => React.ReactElement | null | undefined;
+    preloader: (() => React.ReactNode) | null | undefined;
```

Comment:
The main fix here is wrapping the function that returns a React element, otherwise it's a function that returns `React.ReactElement | null | undefined`. I changed the return type to `React.ReactNode` as that is what a `render` function returns.